### PR TITLE
SequenceK add mapNotNull and deprecate filterMap

### PR DESCRIPTION
**Pull request by tapegram**

### CODE REVIEW

**Review by nomisRev — APPROVED** [review]
🙌

File: arrow-core-data/src/test/kotlin/arrow/core/SequenceKTest.kt [hunk 1]
```diff
@@ -156,6 +156,54 @@ class SequenceKTest : UnitSpec() {
         }
       }
     }
+
+    "filterMap" {
+      forAll(Gen.sequenceK(Gen.int())) { a ->
+        val result = a.asSequence().k()
+          .filterMap {
```
Comment (rachelcarmena) [inline arrow-core-data/src/test/kotlin/arrow/core/SequenceKTest.kt:163]:
```suggestion
        val result =
          a.filterMap {
```

File: arrow-core-data/src/test/kotlin/arrow/core/SequenceKTest.kt [hunk 2]
```diff
@@ -156,6 +156,54 @@ class SequenceKTest : UnitSpec() {
...
+            .asSequence()
+            .k()
+
+        result.toList() == expected.toList()
+      }
+    }
+
+    "mapNotNull" {
+      forAll(Gen.list(Gen.int())) { a ->
+        val result = a.asSequence().k().mapNotNull {
```
Comment (rachelcarmena) [inline arrow-core-data/src/test/kotlin/arrow/core/SequenceKTest.kt:187]:
```suggestion
      forAll(Gen.sequenceK(Gen.int())) { a ->
        val result = a.mapNotNull {
```

**Review by rachelcarmena — APPROVED** [review]
Thank you so much @tapegram ! Just a few small proposals :raised_hands:

**Review by raulraja — APPROVED** [review]

**Pull request merged**